カレンダー問題の自己レビュー後の再提出です

rira100000000 opened the pull request:
# 内容
* option_parseメソッドとメインの処理に対してリファクタリングをしました。
* RuboCopを通るように修正しました。

# 背景
* レビュー前ですが自分で見直してリファクタリングの余地があると感じたので修正しました。
* RuboCopのプラクティス後だったのでRuboCopを通るように修正しました。

rira100000000 committed:
calendar問題 FirstCommit

rira100000000 committed:
リファクタリング

option_parseメソッドの処理をget_optionメソッドと分離

rira100000000 committed:
リファクタリング

メインの処理の一部をprint_calendarメソッドに分離した。

rira100000000 committed:
RuboCopの指摘修正

RuboCopの私的部分を修正した。
print_spaceの処理の一部をposition_calcに分離した。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+require 'date'
+require 'optparse'
+
+FIRST = 1
```
単なる数字に意味のある名前を付けて定数とするのは良い事なのですが、
FIRST/LAST/ZERO_POSITION/MAX_1DIGIT_DAYなどは 数値のみで意味が通じるので不要です。むしろ 「ZERO_POSTIONは0では無いがゼロ位置なので定義しているのかな？」など考え過ぎる要因となります。
その部分が数字なのか定数なのか、どちらが意味が分かりやすいか？を判断してください。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+end
+
+def saturday?(youbi_index)
+  youbi_index % WEEKS_FIRST_YOUBI == SATURDAY
```
WEEKS_FIRST_YOUBI とありますが、一週間の日数を意図しているように見え
定数名を変えた方が良さそうですね。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+  result
+end
+
+def option_parse
```
すごく細かいので参考までですが `parse_option`と動詞_名詞の順番の方が一般的です。
OptionParserから来たのかもしれませんが、OptionParserはこれ全体で名詞にしているのでこの語順です。クラス名は名詞なのでこうなっています。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+  print_space(first_youbi)
+  youbi_index = first_youbi
+  (FIRST..last_day).each do |day|
+    print ' ' if day <= MAX_1DIGIT_DAY # 1~9日はスペースを追加
```
ここについては Stringクラスに便利なメソッドがあって必要無くなるので調べてみてください 💪

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+year = Date.today.year if year.nil?
+month = Date.today.month if month.nil?
+
+# 対象の月の1日を取得
```
ここから3つの変数は全て year/monthが分かれば算出できて、print_calendarでしか使わないものなので print_calendar内で算出した方が良いですね。
変数のスコープ(見える範囲)はできるだけ狭い方が分かりやすくなります。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+end
+
+# 一週目の1日までの間にスペースを置く
+def print_space(first_youbi)
```
コメントが無ければ「空白を出力する」としか取れず、呼び出し元では何をするのか想像できません。
内容を全て含む必要はありませんが、「なんのために空白を出力するのか」を考えて適切な名前を考えてみましょう。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+  if youbi == SUNDAY
+    ZERO_POSITION
+  else
+    youbi * SPACE_NUM
```
SPACE_NUMはスペースの数だとちょっと情報量が少な過ぎると思います。
なんのためのスペースなのかが連想できる名前にしましょう。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+SATURDAY = 6
+MAX_1DIGIT_DAY = 9
+
+def position_calc(youbi)
```
これも基本の語順は calc_positionになりますが、何の位置なのかの情報がもうちょっと欲しいですね。
適切な名前が思いつかないなら一行レベルですしメソッドにしない方が分かりやすいかもしれません。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,107 @@
+  print space
+end
+
+def print_colored_day(day)
```
色んなパターンで色の付いた日を出力するならこの名前でも良いのですが、今回は1パターンだけですし「今日を出力する」という意味にした方が分かりやすいと思います。

rira100000000 committed:
不要な定数を数値に変更

数値のままで十分意味が通る定数は数値に変更しました。
初回レビュー指摘点です。

rira100000000 committed:
定数名、メソッド名を変更

定数名、メソッド名をより実態に即したものに変更しました。
メソッド名が動詞から始まっていないものを修正しました。
初回レビュー指摘点です。

rira100000000 committed:
calc_youbi_positionを削除

1行かつ再利用しないメソッドだったため削除しました。
初回レビュー指摘点です。

rira100000000 committed:
print_calendarとprint_dayの修正

print_calendarメソッド内でのみ使う変数はメソッド内で計算するように修正しました。
print_calendarの処理の一部をrjustメソッドに置き換えました。
また、print_dayの引数をint型に統一したかったため、rjustはprint_day内で実行するようにしました。
初回レビュー指摘点です。

u1tnk commented on 02.calendar/calendar.rb:
```diff
@@ -0,0 +1,95 @@
+    print "\n" if saturday?(youbi_index)
+    youbi_index += 1
+  end
+  puts '' # zshがつける不要な%を消す
```
このコメントはぱっと見なぜ出力しているか分からないのであって良いですね。

https://twitter.com/t_wada/status/904916106153828352

ここには why notとか書いてますが、コミットと同じくwhyでも良いです。
要はコード見ても読み取れない情報を書きましょう。

u1tnk reviewed (approved):
だいぶスッキリしました！

変数名に改善の余地はありますが、明らかに情報が無いなどの問題は無いので合格です。
合格後に他の人のを見て参考にしてみてください。

また、ちょっと指摘が漏れていましたし、細かいので今後の課題で良いですがコメントが多過ぎますね。
一つ、このコメントは必要ってところにレビューコメント付けましたが、それ以外は全部消して良いです。
ソースコード見れば分かる内容のコメントがあると以下の弊害があるので必要なときのみにしましょう。

- ノイズがあって読みづらくなる
- ソースとコメントが微妙に違うときに誤解する
- 修正したときにコメントだけ残ってしまう事が多い

rira100000000 committed:
コメントの削除

コメントの量が多かったため、なくても意味が分かる場所は削除しました。
再レビューの指摘点です。
